tools/pubsys/src/aws/validate_ami/results.rs

```diff
@@ -0,0 +1,1028 @@
+//! The results module owns the reporting of EC2 image validation results.
+
+use super::ami::{ImageDef, Result};
+use aws_sdk_ec2::Region;
+use serde::{Deserialize, Serialize};
+use std::collections::{HashMap, HashSet};
+use std::fmt::{self, Display};
+use std::str::FromStr;
+use tabled::{Table, Tabled};
+
+/// Represent the possible status of an EC2 image validation
+#[derive(Debug, Eq, Hash, PartialEq, Serialize, Deserialize)]
+pub(crate) enum AmiValidationResultStatus {
+    /// The AMI was found and its monitored fields have the expected values
+    Correct,
+
+    /// The AMI was found but some of the monitored fields do not have the expected values
+    Incorrect,
+
+    /// The image was expected but not included in the actual images
+    Missing,
+}
+
+impl Display for AmiValidationResultStatus {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            AmiValidationResultStatus::Correct => write!(f, "Correct"),
+            AmiValidationResultStatus::Incorrect => write!(f, "Incorrect"),
+            AmiValidationResultStatus::Missing => write!(f, "Missing"),
+        }
+    }
+}
+
+impl FromStr for AmiValidationResultStatus {
+    type Err = super::Error;
+
+    fn from_str(s: &str) -> std::result::Result<Self, Self::Err> {
+        match s {
+            "Correct" => Ok(Self::Correct),
+            "Incorrect" => Ok(Self::Incorrect),
+            "Missing" => Ok(Self::Missing),
+            filter => Err(Self::Err::InvalidStatusFilter {
+                filter: filter.to_string(),
+            }),
+        }
+    }
+}
+
+/// Represents a single EC2 image validation result
+#[derive(Debug, Eq, Hash, PartialEq, Serialize)]
+pub(crate) struct AmiValidationResult {
+    /// The id of the image
+    pub(crate) id: String,
+
+    /// ImageDef containing expected values for the image
```

Comment:
nit:
```suggestion
    /// `ImageDef` containing expected values for the image
```

and where ever else this may apply.